xbmc/settings/DisplaySettings.cpp

```diff
@@ -548,23 +548,23 @@ void CDisplaySettings::ApplyCalibrations()
 void CDisplaySettings::UpdateCalibrations()
 {
   CSingleLock lock(m_critical);
-  for (size_t res = RES_DESKTOP; res < m_resolutions.size(); ++res)
+
+  // Add new (unique) resolutions
+  for (ResolutionInfos::iterator res(m_resolutions.begin() + RES_DESKTOP + 1); res != m_resolutions.end(); ++res)
```

Comment:
Always use the const versions of begin and end when possible (here it is).
```suggestion
  for (ResolutionInfos::const_iterator res(m_resolutions.cbegin() + RES_DESKTOP + 1); res != m_resolutions.cend(); ++res)
```